src/manager.cpp

```
@@ -130,9 +130,7 @@ sdbusplus::async::task<> Manager::startSyncEvents()
 void Manager::syncData(const config::DataSyncConfig& dataSyncCfg)
 {
     using namespace std::string_literals;
-    std::string syncCmd{"rsync --archive --compress"};
-
-    // Add source data path
+    std::string syncCmd{"rsync --archive --compress --ignore-missing-args"};
```

> --ignore-missing-args

Why do we need this option? I couldn’t find any details in the commit message.

Also, if we are making changes to the rsync CLI, I would prefer them to be in a separate commit for better clarity.

This option suppresses errors when a file is missing, but I believe we should treat that as an error instead.

Are you encountering any errors during periodic sync when a file is missing? If so, we should handle it specifically in the periodic logic. One approach could be to generate an error and stop the periodic sync for that particular file. However, data sync is triggered automatically during BMC boot after the role is determined. At the same time, several other applications might start and configure their data in the data sync configuration to sync.

If data sync starts before those applications are starts, we should not treat this as an error. Instead, we should log a warning or debug message and let the process sleep according to the configured periodicity or we could use inotify framework to watch its creation and sync it then go for sleep as per configured interval for Periodic type sync.

---

If our source path is not present, the rsync command fails. To suppress it was using 'ignore-missing-args'. 
Will let the rsync to fail and get logged. 
Could be handled in future with --delete option and error handling based on [error code](https://stackoverflow.com/questions/20737204/comprehensive-list-of-rsync-error-codes).